LayoutKitTests/OverlayLayoutTests.swift

```diff
@@ -152,30 +157,59 @@ class OverlayLayoutTests: XCTestCase {
-                        let frame = Constant.alignments[index].position(size: backgroundSize, in: expectedRect)
-                        AssertEqualDensity(subLayout.frame, [1.0: frame, 2.0: frame, 3.0: frame])
-                    } else if index == backgroundCount + 1 {
-                        // We have the primary layout - should be laid out just like the overlay itself
-                        AssertEqualDensity(subLayout.frame, [1.0: expectedRect, 2.0: expectedRect, 3.0: expectedRect])
-                    } else if index > backgroundCount + 1 {
-                        // We have an overlay layout
-                        let overlaySize = CGSize(width: Measurement.overlayWidth, height: Measurement.overlayHeight)
-                        let frame = Constant.alignments[index - backgroundCount - 2].position(
-                            size: overlaySize,
-                            in: expectedRect)
-                        AssertEqualDensity(subLayout.frame, [1.0: frame, 2.0: frame, 3.0: frame])
-                    }
-                }
+    /**
+     Makes sure that huge overlays and underlays don't overflow their bounds - they are constrained
+     to the size of the primary layout.
+     */
+    func testHugeOverlaysAndUnderlays() {
+        Constant.alignments.forEach { primaryAlignment in
+            Constant.alignments.forEach { hugeAlignment in
+                let huge = SizeLayout(
+                    width: Measurement.hugeWidth,
+                    height: Measurement.hugeHeight,
+                    alignment: hugeAlignment)
+                let primary = SizeLayout(
+                    width: Measurement.primaryWidth,
+                    height: Measurement.primaryHeight,
+                    alignment: primaryAlignment)
+                let overlay = OverlayLayout(primary: primary, background: [huge], overlay: [huge])
+                let expectedFrame = CGRect(
+                    origin: .zero,
+                    size: CGSize(width: Measurement.primaryWidth, height: Measurement.primaryHeight))
+                let arrangement = overlay.arrangement()
+                AssertEqualDensity(arrangement.frame, [1.0: expectedFrame, 2.0: expectedFrame, 3.0: expectedFrame])
+            }
+        }
+    }
+
+    /**
+     Tests the overlay layout when the size it is being laid into is unlimited. Confirms that all the
+     alignments are correct, using a complex layout with lots of sublayouts. Lays it out into the top
+     left so we can know the correct size.
+     */
+    func testUnlimitedSize() {
+        let layoutSize = CGSize(width: Measurement.primaryWidth, height: Measurement.primaryHeight)
+        let expectedRect = CGRect(origin: .zero, size: layoutSize)
+        (0..<Constant.alignments.count).forEach { backgroundCount in
+            (0..<Constant.alignments.count).forEach { overlayCount in
```

Comment:
Is it worth it iterating over the different counts? Seems like the last iteration is the union of all of them.